codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/RustWriter.kt

```diff
@@ -305,6 +305,12 @@ fun Writable.isEmpty(): Boolean {
     return writer.toString() == RustWriter.root().toString()
 }
 
+/**
+ * Rustdoc doesn't support `r#` for raw identifiers.
+ * This function adjusts doc links to refer to raw identifiers directly.
+ */
+fun docLink(docLink: String): String = docLink.replace("::r##", "::").replace("::r#", "::")
```

Comment:
~there _may_ be other places where we make doc links that need to be audited (but maybe those aren't dynamic?)~

Nevermind, looks like you got em